src/http/client.cc

```diff
@@ -141,8 +141,17 @@ future<connection::reply_ptr> connection::recv_reply() {
                 throw std::system_error(ECONNABORTED, std::system_category());
             }
             if (parser.failed()) {
-                http_log.trace("Parsing response failed");
-                throw std::runtime_error("Invalid http server response");
+                http_log.warn("Parsing response failed.{}", http_log.level() == log_level::debug ? format(" Reason: {}", parser.error_message()) : "");
+                // Parsing the response failed, potentially due to a server-side bug or
+                // a transient error. Retrying might resolve the issue. To trigger a retry,
+                // return EPIPE to the caller. The original exception is retained as a
+                // nested exception so that, in case retries keep failing, the caller can
+                // identify the actual cause unrelated to the EPIPE error.
```

Comment:
How would caller identify the actual error if it's going to be system_error anyway? Also, the error message is not in the exception, so the caller is still blind.